tests/CMakeLists.txt

```diff
@@ -38,67 +38,30 @@ add_subdirectory(${CMAKE_BINARY_DIR}/googletest-src
 string(REGEX REPLACE "-Werror[^ ]*" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
 string(REPLACE "-pedantic-errors" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
 
-function(unit_test file tests)
-  set(multi_value_args SOURCES)
-
-  cmake_parse_arguments("BIN" "" "" "${multi_value_args}" ${ARGN})
-
-  # Prefix all sources needed by the tests with ../src/ so that the calls to the
-  # unit_test function become cleaner
-  SET(sources "")
-  FOREACH(f ${BIN_SOURCES})
-    # Do not add main.cpp, because it will override the main function
-    if(NOT "${f}" STREQUAL "main.cpp")
-      LIST(APPEND sources "../src/${f}")
-    endif()
-  ENDFOREACH(f)
+# Compile all unit tests with 'make all_unit_tests'
+add_custom_target(all_unit_tests
+    COMMENT "Building all unit test")
 
-  string(REPLACE "/" "_" testname ${file})
+function(add_unit_test source_file)
```

Comment:
Do you mean the name change or the actual refactor of the CMake function?

If you meant the change in logic: I tried to add a test for config.cpp, but I was getting a lot of link errors due to missing symbols. Some of them were from functions defined in `src`, but others were from symbols defined by entries in the `libs` variable. Instead of selectively recompiling some source files in order to build the unit tests, I split the polybar executable into a library and an executable [here](https://github.com/sam-lunt/polybar/commit/e2605aea2d4607a61d7590e160dcd88fd6019c9b#diff-95e351a3805a1dafa85bf20b81d086e6R72), which allows the tests to link against `polybar-lib` (if you have a better name, I'm not the biggest fan of that), which includes the dependencies as well as the symbols in `src`. In addition to making the unit test logic simpler, it should improve build times (not that that's a big concern).

If you meant the name, I changed it because I thought `unit_test` was a little unclear, and `add_*` fits with the other CMake functions `add_executable`, `add_library`, `add_test`, etc.

Reply:
Oh, I see. Originally, the idea was to have each test specify all source file it relied on so that tests didn't need to compile every polybar source file. But some `.cpp` file are not decoupled at all and need to be compiled with nearly all other source file, that's why I used the `${files}` variable in those tests.

So yeah, this solution is a lot cleaner and does reduce compile times when the whole suite is compiled.

Maybe we can call it `poly`, so the library generated is `libpoly`.

`add_unit_test` is fine